handler/src/main/java/io/netty5/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -517,57 +528,89 @@ private int writePlaintextData(final ByteBuffer src, int len) {
                 src.position(pos + sslWrote);
             }
         } else {
-            ByteBuf buf = alloc.directBuffer(len);
-            try {
-                src.limit(pos + len);
-
-                buf.setBytes(0, src);
-                src.limit(limit);
-
-                sslWrote = SSL.writeToSSL(ssl, memoryAddress(buf), len);
+            try (Buffer buf = alloc.allocate(len)) {
+                buf.writeBytes(src.array(), src.arrayOffset() + pos, len);
+                SslWrite write = new SslWrite(ssl, len);
```

Comment:
couldn't we re-use the same instance ?

Reply:
Right, the engine is not accessed concurrently.